### PR TITLE
hack,Makefile: Update the deploy scripts to use the deploy-metering binary.

### PR DESCRIPTION
**Pull request by timflannagan1**
This is based on top of #940, which adds a CLI for deploying metering resources.

In this PR, all the `./hack/*install.sh` and `./hack/*uninstall.sh` scripts are replaced (except for olm) in the deploy scripts and now use the `deploy-metering` binary to manage the metering resources. Also, the `./hack/run-metering-operator-local.sh` now calls the `./hack/lib/util.sh install_metering()` helper function to install metering resources. 

At this moment, only Openshift/OCP and OLM installs are tested in e2e/integration suite. In the future, we can think about testing upstream installs (e.g. `install_metering upstream`) and alter the `install_metering` helper function to call `./bin/deploy-metering install --platform upstream` (or `export DEPLOY_PLATFORM="upstream"` and forgo the platform flag.)

### CODE REVIEW

**Comment by timflannagan1** [comment]
Oh yeah forgot to build the binary.

File: hack/deploy.sh [hunk 1]
```diff
@@ -8,7 +8,8 @@ source "${ROOT_DIR}/hack/common.sh"
 
 : "${UNINSTALL_METERING_BEFORE_INSTALL:=true}"
 : "${INSTALL_METERING:=true}"
-: "${INSTALL_METHOD:=openshift-direct}"
+: "${INSTALL_METHOD:=generic}"
+: "${UNINSTALL_METHOD:=generic}"
 : "${METERING_CREATE_PULL_SECRET:=false}"
```
Comment (EmilyM1) [inline hack/deploy.sh]: What about naming it "openshift" instead of "generic"?

File: hack/run-metering-operator-local.sh [hunk 2]
```diff
@@ -6,7 +6,7 @@ source "${ROOT_DIR}/hack/common.sh"
 
 : "${METERING_OPERATOR_IMAGE:=${METERING_OPERATOR_IMAGE_REPO}:${METERING_OPERATOR_IMAGE_TAG}}"
 : "${LOCAL_METERING_OPERATOR_RUN_INSTALL:=true}"
-: "${METERING_INSTALL_SCRIPT:=./hack/openshift-install.sh}"
+: "${INSTALL_METHOD:=generic}"
```
Comment (chancez) [inline hack/run-metering-operator-local.sh]: Should default to openshift no?

**Comment by chancez** [comment]
/hold

So far this seems fine, but we need to be very careful to not break e2e/integration. I'd like to hold on this until 4.2 freeze is done, then we can merge without affecting 4.2 builds/CI.

File: hack/run-metering-operator-local.sh (hunk 2 above)
Comment (timflannagan1) [reply hack/run-metering-operator-local.sh]: Fixed.

**Comment by timflannagan1** [comment]
1.16 rebase is done
/hold cancel

**Pull request merged**